### PR TITLE
Fixing bugs in nGraph reference implementation of Interpolate-4

### PR DESCRIPTION
**Pull request by vgavrilo**
Fixing bugs in nGraph reference implementation of Interpolate-4

Issue #39093

### CODE REVIEW

File: ngraph/test/op_eval/interpolate.cpp [hunk 1]
```diff
@@ -14,6 +14,7 @@
 // limitations under the License.
 //*****************************************************************************
 
+#include <fstream>
```
Comment (lazarevevgeny) [inline ngraph/test/op_eval/interpolate.cpp:17]: please, revert changes in this file.

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/skip_tests_config.cpp [hunk 2]
```diff
@@ -50,6 +50,7 @@ std::vector<std::string> disabledTestPatterns() {
         // TODO: Issue: 32032
         R"(.*ActivationParamLayerTest.*)",
         // TODO: Issue: 37862
-        R"(.*ReverseSequenceLayerTest.*netPRC=(I8|U8).*)"
+        R"(.*ReverseSequenceLayerTest.*netPRC=(I8|U8).*)",
+        R"(.*InterpolateLayerTest.*)"
```
Comment (lazarevevgeny) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/skip_tests_config.cpp:54]: There are still failing tests in the pre-commit. Please, disable them.

File: ngraph/core/reference/src/runtime/reference/interpolate.cpp [hunk 3]
```diff
@@ -39,18 +39,19 @@ std::array<float, 4> InterpolateEvalHelper::get_cubic_coeff(float s, float a)
...
+        float length_original = static_cast<float>(m_input_data_shape[i]);
+        float in_coord = m_get_original_coord(static_cast<float>(output_coord[i]),
+                                              m_all_scales[i],
+                                              static_cast<float>(m_out_shape[i]),
                                               length_original);
-        int64_t nearest_pixel = m_get_nearest_pixel(in_coord, m_scales[i] < 1.0);
-        input_coord[axis] = clip_coord(nearest_pixel, length_original);
+        int64_t nearest_pixel = m_get_nearest_pixel(in_coord, m_all_scales[i] < 1.0);
+        input_coord[i] =  std::max(static_cast<int64_t>(0),
+                                   std::min(nearest_pixel,
```
Comment (lazarevevgeny) [inline ngraph/core/reference/src/runtime/reference/interpolate.cpp:53]: Please, fix the code style by applying patch from https://github.com/openvinotoolkit/openvino/suites/1195651754/artifacts/17638098

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: ngraph/core/reference/src/runtime/reference/interpolate.cpp (hunk 3 above)
Comment (vgavrilo) [reply ngraph/core/reference/src/runtime/reference/interpolate.cpp]: Fixed.

File: ngraph/test/op_eval/interpolate.cpp (hunk 1 above)
Comment (vgavrilo) [reply ngraph/test/op_eval/interpolate.cpp]: Fixed.

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/skip_tests_config.cpp (hunk 2 above)
Comment (vgavrilo) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/skip_tests_config.cpp:54]: Tests were disabled.

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/skip_tests_config.cpp [hunk 4]
```diff
@@ -50,6 +50,7 @@ std::vector<std::string> disabledTestPatterns() {
         // TODO: Issue: 32032
         R"(.*ActivationParamLayerTest.*)",
         // TODO: Issue: 37862
-        R"(.*ReverseSequenceLayerTest.*netPRC=(I8|U8).*)"
+        R"(.*ReverseSequenceLayerTest.*netPRC=(I8|U8).*)",
```
Comment (lazarevevgeny) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/skip_tests_config.cpp:53]: Enable tests.

File: ngraph/test/op_eval/interpolate.cpp [hunk 5]
```diff
@@ -556,4 +556,4 @@ TEST(op_eval, interpolate_v4_linear_onnx)
         }
         ++i;
     }
-}
+}
```
Comment (lazarevevgeny) [inline ngraph/test/op_eval/interpolate.cpp:559]: revert this change

File: inference-engine/tests/functional/plugin/shared/src/single_layer_tests/interpolate.cpp [hunk 6]
```diff
@@ -86,7 +86,7 @@ void InterpolateLayerTest::SetUp() {
     function = std::make_shared<ngraph::Function>(results, params, "interpolate");
 }
 
-TEST_P(InterpolateLayerTest, CompareWithRefs) {
+TEST_P(InterpolateLayerTest, DISABLED_CompareWithRefs) {
```
Comment (lazarevevgeny) [inline inference-engine/tests/functional/plugin/shared/src/single_layer_tests/interpolate.cpp:89]: Enable tests.

Comment (vgavrilo) [reply inference-engine/tests/functional/plugin/shared/src/single_layer_tests/interpolate.cpp]: Done.

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/skip_tests_config.cpp (hunk 4 above)
Comment (vgavrilo) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/skip_tests_config.cpp:53]: Done.

File: ngraph/test/op_eval/interpolate.cpp (hunk 5 above)
Comment (vgavrilo) [reply ngraph/test/op_eval/interpolate.cpp]: Done.

**Review by dmitry-gorokhov — APPROVED** [review]
CPU part looks good to me

**Review by lazarevevgeny — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]
